libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/infopanel/InfoPanelEndNavigationButtonBinder.kt

```diff
@@ -3,20 +3,29 @@ package com.mapbox.navigation.dropin.infopanel
 import android.view.ViewGroup
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.dropin.R
 import com.mapbox.navigation.dropin.extendablebutton.EndNavigationButtonComponent
+import com.mapbox.navigation.dropin.internal.extensions.reloadOnChange
+import com.mapbox.navigation.dropin.internal.extensions.updateMargins
 import com.mapbox.navigation.dropin.navigationview.NavigationViewContext
 import com.mapbox.navigation.ui.base.lifecycle.UIBinder
+import com.mapbox.navigation.ui.base.view.MapboxExtendableButton
 
 @ExperimentalPreviewMapboxNavigationAPI
 internal class InfoPanelEndNavigationButtonBinder(
     private val context: NavigationViewContext
 ) : UIBinder {
 
     override fun bind(viewGroup: ViewGroup): MapboxNavigationObserver {
-        return EndNavigationButtonComponent(
-            context.store,
-            viewGroup,
-            context.styles.endNavigationButtonParams
-        )
+        return reloadOnChange(context.styles.endNavigationButtonStyle) { style ->
+            val button = MapboxExtendableButton(viewGroup.context, null, 0, style)
+            viewGroup.removeAllViews()
+            viewGroup.addView(button)
+            button.updateMargins(
+                right = button.resources.getDimensionPixelSize(R.dimen.mapbox_infoPanel_paddingEnd)
```

Comment:
We could also add `left` even though it wouldn't produce any visible effect, but would fail if this button is moved to the left as a part of RTL or design change.

Reply:
Let's address this later when we add RTL support across the entire NavigationView.